src/execution_stats.rs

```diff
@@ -0,0 +1,98 @@
+///! Maintenance of execution stats.
+///!
+///! The API models TimingInformation as optional, which can make it a little
+///! tricky to accumulate processing time (consider fetching pages of results,
+///! summing total time). The traits here serve to organize this concern out of
+///! the main body of code.
+use rusoto_qldb_session::{IOUsage, TimingInformation};
+
+pub trait TimingInformationExt {
+    fn accumulate(&mut self, other: &Option<TimingInformation>);
+}
+
+impl<T> TimingInformationExt for Option<T>
+where
+    T: TimingInformationExt,
+{
+    fn accumulate(&mut self, other: &Option<TimingInformation>) {
+        if let Some(t) = self {
+            t.accumulate(other);
+        }
+    }
+}
+
+impl TimingInformationExt for TimingInformation {
+    fn accumulate(&mut self, other: &Option<TimingInformation>) {
+        if let Some(o) = other {
+            if let (Some(i), Some(j)) = (
+                self.processing_time_milliseconds,
+                o.processing_time_milliseconds,
+            ) {
+                self.processing_time_milliseconds = Some(i + j)
+            }
+        }
+    }
+}
+
+pub trait IOUsageExt {
+    fn accumulate(&mut self, other: &Option<IOUsage>);
+}
+
+impl<T> IOUsageExt for Option<T>
+where
+    T: IOUsageExt,
+{
+    fn accumulate(&mut self, other: &Option<IOUsage>) {
+        if let Some(t) = self {
+            t.accumulate(other);
+        }
+    }
+}
+
+impl IOUsageExt for IOUsage {
+    fn accumulate(&mut self, other: &Option<IOUsage>) {
+        if let Some(o) = other {
+            if let (Some(i), Some(j)) = (self.read_i_os, o.read_i_os) {
+                self.read_i_os = Some(i + j)
+            }
+
+            if let (Some(i), Some(j)) = (self.write_i_os, o.write_i_os) {
+                self.write_i_os = Some(i + j)
+            }
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn accumulating_stats() {
+        let mut stats = (
+            TimingInformation {
+                processing_time_milliseconds: Some(1),
+            },
+            IOUsage {
+                read_i_os: Some(1),
```

Comment:
I would choose different numbers here just to be sure we don't add reads to writes or something like that

Reply:
Good suggestion. Also, you merged this PR into the parent branch (which wasn't main), so I'll make the change, reopen the PR and set the parent branch to main.

(The reason the parent branch wasn't main was because this change depended on the upgraded SDK.)

Reply:
https://github.com/awslabs/amazon-qldb-driver-rust/pull/17 fixes this.